### DISCUSSION TITLE
DS3231(m) support

### DISCUSSION THREAD

**Discussion by erlendnagel** [paulscottrobson/neo6502-firmware#372] [Q&A] [answered]
I happen to have a low-cost DS3231m RTC clock. It is not the Olimex UEXT module, but a cheap Chinese version: Still it works ok.
I would like to contribute the code to support that next to the existing PCF 8563 UEXT Library.

Should I just provide the same 4 operations:
rtc.readtime(ref h,ref m,ref s)
rtc.readdate(ref y,ref m,ref d)
rtc.writetime(h,m,s)
rtc.writedate(y,m,d)
Note: I'm not sure if rtc.read and rtc.write would work since some of the date/time fields are a little awkward. 
Or should I introduce new names? Theoretically it should be possible to have two I2C RTC clocks because the PCF8563 and DS3231 have different ID's ($51 and $68), but I'm not sure it is useful. Rather it would make more sense for the code to figure out and set sensible default settings for e.g. use of the 24hr clock automatically, by having the library check which, if any clock is there.

**Comment by paulscottrobson** [comment]
It's entirely up to you :)  If you want to revamp the whole thing that's fine ....

**Comment by erlendnagel** [comment, marked as answer]
Thank you. I have made changes to support additional models and submitted the updated code. Let me know in case you spot issues with the code or in case I didn't submit in the right way (I am a newbie when it comes to git)

**Comment by erlendnagel** [comment]
Implemented
